cis194/week2/zehua/test/ExercisesSpec.hs

```diff
@@ -0,0 +1,59 @@
+module ExercisesSpec where
+
+import Test.Hspec
+import Exercises
+import Log
+
+spec :: Spec
+spec = do
+  describe "ex1.parseMessage" $ do
+    it "works with error" $ do
+      parseMessage "E 2 562 help help" `shouldBe`
+        LogMessage (Error 2) 562 "help help"
+
+    it "parses error with out-of-range severity as unknown" $ do
+      parseMessage "E 0 562 help help" `shouldBe`
+        Unknown "E 0 562 help help"
+      parseMessage "E 101 562 help help" `shouldBe`
+        Unknown "E 101 562 help help"
+
+    it "works with warn" $ do
+      parseMessage "W 567 lu la la" `shouldBe` LogMessage Warning 567 "lu la la"
+
+    it "works with info" $ do
+      parseMessage "I 29 la la la" `shouldBe` LogMessage Info 29 "la la la"
+
+    it "works with invalid timestamp" $ do
+      parseMessage "W abc This is not in the right format" `shouldBe`
+        Unknown "W abc This is not in the right format"
+
+    it "works with unknown" $ do
+      parseMessage "This is not in the right format" `shouldBe`
+        Unknown "This is not in the right format"
+
+  describe "ex1.parse" $ do
+    it "works with sample" $ do
+      actual <- testParse parse 11 "testdata/sample.log"
+      actual `shouldBe` [
+        LogMessage Info 6 "Completed armadillo processing",
+        LogMessage Info 1 "Nothing to report",
+        LogMessage Info 4 "Everything normal",
+        LogMessage Info 11 "Initiating self-destruct sequence",
+        LogMessage (Error 70) 3 "Way too many pickles",
+        LogMessage (Error 65) 8 "Bad pickle-flange interaction detected",
+        LogMessage Warning 5 "Flange is due for a check-up",
+        LogMessage Info 7 "Out for lunch, back in two time steps",
+        LogMessage (Error 20) 2 "Too many pickles",
+        LogMessage Info 9 "Back from lunch",
+        LogMessage (Error 99) 10 "Flange failed!"
+        ] 
```

Comment:
forgot to ask whether this way of formatting lists is the recommended way. @shineli1984

Reply:
The convention is 
```Haskell
[ element1
, element2
, element3 ]
```